molo/core/tag_navigation.py

```diff
@@ -0,0 +1,209 @@
+
+            if sec_articles_for_locale and len(
+                    sec_articles_for_locale) > self.sec_articles_count:
+                sec_articles_for_locale = sec_articles_for_locale[
+                    :self.sec_articles_count]
+
+            if section_for_locale:
+                sections_list.append((
+                    section,
+                    sec_articles_for_locale))
+
+        return sections_list
+
+    def get_cached_sections_list(self):
+        cache_key = self.get_sections_list_cache_key()
+        sections_list_cache = cache.get(cache_key)
+        if sections_list_cache:
+            return [
+                (SectionPage.objects.get(pk=section_pk),
+                 ArticlePage.objects.filter(pk__in=articles_pk).order_by(
+                    '-featured_in_homepage_start_date').live())
+                for section_pk, articles_pk in sections_list_cache]
+
+    def get_tags_article_list(self):
+        tags_list = []
+        tag_qs = Tag.objects.descendant_of(self.request.site.root_page).filter(
+            feature_in_homepage=True).live()
+        for tag in tag_qs:
+            tag_articles = get_articles_for_tags_with_translations(
+                self.request, tag, self.locale,
+                self.context, self.exclude_pks)
+            if tag_articles and len(tag_articles) > self.tag_count:
+                tag_articles = tag_articles[:self.tag_count]
+            self.exclude_pks += [p.pk for p in tag_articles]
+            tag_for_locale = get_pages(
+                self.context, tag_qs.filter(pk=tag.pk), self.locale)
+            if tag_for_locale:
+                tags_list.append((tag_for_locale[0], tag_articles))
+            else:
+                tags_list.append((tag, tag_articles))
+        return tags_list
+
+    def get_cached_tags_article_list(self):
+        cache_key = self.get_tags_article_list_cache_key()
+        tags_article_list_cache = cache.get(cache_key)
+        if tags_article_list_cache:
+            return [
+                (Tag.objects.get(pk=tag_pk),
+                 ArticlePage.objects.filter(pk__in=articles_pk).order_by(
+                    '-featured_in_homepage_start_date').live())
+                for tag_pk, articles_pk in tags_article_list_cache]
+
+    def get_data(self):
+        data = {}
+
+        latest_articles_cache = self.get_cached_featured_latest_articles()
+        latest_articles = latest_articles_cache
+        if latest_articles_cache is None:
+            latest_articles = self.get_featured_latest_articles()
+
+        # Featured Section/s
+        sections_list = self.get_cached_sections_list()
+        if not sections_list:
+            sections_list = self.get_sections_articles_list()
+            cache.set(
+                self.get_sections_list_cache_key(),
+                [(section.pk, [article.pk for article in articles])
+                 for section, articles in sections_list],
+                300)
```

Comment:
can we maybe change the expiry time to something more than 5 min, maybe every hour or every day?